sdk/src/client/api/block_builder/transaction_builder/mod.rs

```diff
@@ -680,38 +672,39 @@ impl TransactionBuilder {
         })
     }
 
-    // Inputs need to be sorted before signing, because the reference unlock conditions can only reference a lower index
-    pub(crate) fn sort_input_signing_data(
-        mut inputs: Vec<InputSigningData>,
-        commitment_slot_index: SlotIndex,
-        committable_age_range: CommittableAgeRange,
-    ) -> Result<Vec<InputSigningData>, TransactionBuilderError> {
-        // initially sort by output to make it deterministic
-        // TODO: rethink this, we only need it deterministic for tests, for the protocol it doesn't matter, also there
-        // might be a more efficient way to do this
-        inputs.sort_by_key(|i| i.output.pack_to_vec());
-        // filter for ed25519 address first
-        let (mut sorted_inputs, account_nft_address_inputs): (Vec<InputSigningData>, Vec<InputSigningData>) =
-            inputs.into_iter().partition(|input_signing_data| {
+    fn insert_input_sorted(&mut self, input: InputSigningData) {
+        let input_required_address = input
+            .output
+            .required_address(
+                self.latest_slot_commitment_id.slot_index(),
+                self.protocol_parameters.committable_age_range(),
+            )
+            // PANIC: safe to unwrap as non basic/account/foundry/nft/delegation outputs are already filtered out.
+            .unwrap()
+            .expect("expiration unlockable outputs already filtered out");
+        if input_required_address.is_ed25519() {
+            if let Some(position) = self
+                .selected_inputs
+                .iter()
+                .position(|i| i.output_id() > input.output_id())
```

Comment:
Why is it comparing the output ids like this?

Reply:
Yaknow, I don't know 😆 I think I got confused